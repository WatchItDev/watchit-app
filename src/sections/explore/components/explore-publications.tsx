import CarouselPosterMini from '@src/components/carousel/variants/CarouselPosterMini.tsx';

import { appId, PublicationType, usePublications } from '@lens-protocol/react-web';

import { useResponsive } from '@src/hooks/use-responsive.ts';
import { ExplorePublicationsSkeleton } from '@src/sections/explore/components/explore-publications.skeleton.tsx';

// ----------------------------------------------------------------------

export const ExplorePublications = () => {
  const lgUp = useResponsive('up', 'lg');
  const { data, loading } = usePublications({
    where: {
      publicationTypes: [PublicationType.Post],
      metadata: {
        publishedOn: [appId('watchit')],
      },
    },
  });

  let minItemWidth = 250;
  let maxItemWidth = 350;

  if (!lgUp) {
    minItemWidth = 170;
    maxItemWidth = 250;
  }

  if (loading) return <ExplorePublicationsSkeleton />;

  return (
    <CarouselPosterMini
      data={data ?? []}
      title="Publications"
      minItemWidth={minItemWidth}
      maxItemWidth={maxItemWidth}
    />
  );
}
