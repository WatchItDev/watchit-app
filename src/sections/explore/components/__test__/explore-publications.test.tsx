import { describe, it, vi, expect } from "vitest";
import { render } from "@testing-library/react";
import { ExplorePublications } from "../explore-publications";

vi.mock("react-redux", () => ({
  useDispatch: () => vi.fn(),
}));

vi.mock("@src/hooks/use-responsive.ts", () => ({
  useResponsive: () => true,
}));

vi.mock("@lens-protocol/react-web", async () => {
  const actual = await import("@lens-protocol/react-web");
  return {
    ...actual,
    appId: (id: string) => id,
    usePublications: () => ({
      data: [],
      loading: false,
    }),
    PublicationType: {
      Post: "Post",
    },
  };
});

describe("Testing in the <ExplorePublications/> component", () => {
  it("should match snapshot", () => {
    const { container } = render(<ExplorePublications />);
    expect(container).toMatchSnapshot();
  });
});
