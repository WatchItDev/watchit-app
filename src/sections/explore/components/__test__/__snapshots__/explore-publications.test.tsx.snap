// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing in the <ExplorePublications/> component > should match snapshot 1`] = `
<div>
  <div
    style="position: relative; width: 100%;"
  >
    <div
      style="position: static; inset: unset; width: 100%; pointer-events: none; opacity: 1;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-19i2hl5-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardHeader-root css-hl0h5w-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              Publications
            </span>
          </div>
        </div>
        <div
          class="MuiCardContent-root css-ceyv38-MuiCardContent-root"
        >
          <div
            class="MuiBox-root css-i6bazn"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <div
                class="MuiBox-root css-rmtq2p"
                data-testid="skeleton-item"
              >
                <div
                  class="MuiBox-root css-168a3lq"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-6uim4l-MuiSkeleton-root"
                    role="progressbar"
                  />
                </div>
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1ly8sje-MuiSkeleton-root"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1yige66-MuiSkeleton-root"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1yige66-MuiSkeleton-root"
                />
              </div>
            </div>
            <div
              class="MuiBox-root css-k008qs"
            >
              <div
                class="MuiBox-root css-rmtq2p"
                data-testid="skeleton-item"
              >
                <div
                  class="MuiBox-root css-168a3lq"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-6uim4l-MuiSkeleton-root"
                    role="progressbar"
                  />
                </div>
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1ly8sje-MuiSkeleton-root"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1yige66-MuiSkeleton-root"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1yige66-MuiSkeleton-root"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style="width: 100%; opacity: 0;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardHeader-root css-1pkklwm-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              Publications
            </span>
          </div>
          <div
            class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="leftButton"
              tabindex="0"
              type="button"
            >
              <svg
                class="tabler-icon tabler-icon-chevron-left "
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 6l-6 6l6 6"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="rightButton"
              tabindex="0"
              type="button"
            >
              <svg
                class="tabler-icon tabler-icon-chevron-right "
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6l6 6l-6 6"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCardContent-root css-fd8ij7-MuiCardContent-root"
        >
          <div
            class="MuiBox-root css-1v61vtn"
          >
            <div
              class="slick-slider slick-initialized"
            >
              <div
                class="slick-list"
                style="height: 0px;"
              >
                <div
                  class="slick-track"
                  style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
